src/base/static/js/views/app-view.js

```diff
@@ -750,45 +719,52 @@ module.exports = Backbone.View.extend({
       self.setBodyClass("content-visible");
       self.showSpotlightMask();
 
-      if (layer) {
-        center = layer.getLatLng
-          ? layer.getLatLng()
-          : layer.getBounds().getCenter();
-        zoom = map.getZoom();
+      const geometry = model.get("geometry");
 
-        self.ensureLayerVisibility(datasetId);
-
-        if (model.get("story")) {
-          if (!model.get("story").spotlight) {
-            self.hideSpotlightMask();
-          }
-          self.isStoryActive = true;
-          self.isProgrammaticZoom = true;
-          self.setStoryLayerVisibility(model);
-          center = model.get("story").panTo || center;
-          zoom = model.get("story").zoom;
+      if (model.get("story")) {
+        self.isStoryActive = true;
+        self.isProgrammaticZoom = true;
+        self.setStoryLayerVisibility(model);
+        if (!model.get("story").spotlight) {
+          self.hideSpotlightMask();
         }
+      }
 
-        if (layer.getLatLng) {
-          map.setView(center, zoom, {
-            animate: true,
-          });
-        } else {
-          // If we've defined a custom zoom for a polygon layer for some reason,
-          // don't use fitBounds and instead set the zoom defined
-          if (model.get("story") && model.get("story").hasCustomZoom) {
-            map.setView(center, model.get("story").zoom, {
-              animate: true,
-            });
-          } else {
-            map.fitBounds(layer.getBounds(), {
-              animate: true,
-            });
-          }
-        }
+      const story = model.get("story") || {};
+      // Determine map settings, reconciling with custom story settings if this
+      // model is part of a story.
+      const coordinates = story.panTo || geometry.coordinates;
+      const zoom = story.zoom || this.mapView.map.getZoom();
+
+      if (geometry.type === "LineString") {
+        // Fit to bounds
+        // https://www.mapbox.com/mapbox-gl-js/example/zoomto-linestring
+        const bounds = coordinates.reduce(
+          (bounds, coord) => bounds.extend(coord),
+          this.mapView.map.makeLngLatBounds(coordinates[0], coordinates[0]),
+        );
+        this.mapView.map.fitBounds(bounds, { padding: 30 });
+      } else if (geometry.type === "Polygon") {
+        const bounds = coordinates[0].reduce(
```

Comment:
Ah good catch-- I think this would have been a bug had we tried to set coordinates on a story object that's a polygon or linestring. I think we should convert the `{ lat, lng }` object from the config into an array.